pkg/app/piped/executor/waitapproval/waitapproval.go

```diff
@@ -161,3 +160,45 @@ func (e *Executor) getMentionedAccounts(event model.NotificationEventType) ([]st
 
 	return notification.FindSlackAccounts(event), nil
 }
+
+// True means that number of approvers is valid
+func (e *Executor) validateApproverNum(ctx context.Context, approver string, minApproverNum int) bool {
+	if minApproverNum <= 1 {
+		if err := e.MetadataStore.Stage(e.Stage.Id).Put(ctx, approvedByKey, approver); err != nil {
+			e.LogPersister.Errorf("Unable to save approver information to deployment, %v", err)
+		}
+		e.LogPersister.Infof("Got approval from %q", approver)
+		e.reportApproved(approver)
+		e.LogPersister.Infof("This stage has been approved by %d user (%s)", minApproverNum, approver)
```

Comment:
nits, in case this `minApproverNum` (which come from `e.StageConfig.WaitApprovalStageOptions.MinApproverNum`) is 0 as default, the text will be `This stage has been approved by 0 user (khanhtc1202)` which a bit weird to me.